File: Ontology/DisjointClasses.cs
using System.Collections.Generic;

namespace Ontology
{
    public class DisjointClasses:
        Axiom,
        IDisjointClasses
    {
        private readonly IList<IClassExpression> _classExpressions;

        public DisjointClasses(
            IOntology                 ontology,
            params IClassExpression[] classExpressions
            ) : base(ontology)
        {
            _classExpressions = classExpressions;
        }

        IList<IClassExpression> IDisjointClasses.ClassExpressions => _classExpressions;
    }
}

File: Web/ClientApp/src/app/Ontology/DisjointClasses.ts
import { Axiom } from "./Axiom";
import { IClassExpression } from "./IClassExpression";
import { IDisjointClasses } from "./IDisjointClasses";
import { IOntology } from "./IOntology";

export class DisjointClasses
    extends Axiom
    implements IDisjointClasses
{
    constructor(
        ontology               : IOntology,
        public ClassExpressions: IClassExpression[]
        )
    {
        super(ontology);
    }
}
